db/hints/sync_point.cc

```diff
@@ -69,14 +71,13 @@ static std::vector<sync_point::shard_rps> decode_one_type_v1(uint16_t shard_coun
 
 sync_point sync_point::decode(sstring_view s) {
     bytes raw = base64_decode(s);
-    if (raw.empty()) {
-        throw std::runtime_error("Could not decode the sync point - not a valid hex string");
-    }
-    if (raw[0] != 1) {
-        throw std::runtime_error(format("Unsupported sync point format version: {}", int(raw[0])));
+    seastar::simple_memory_input_stream in{reinterpret_cast<const char*>(raw.data()), raw.size()};
+
+    uint8_t version = ser::serializer<uint8_t>::read(in);
```

Comment:
The `raw.empty()` check is gone now. What will happen when it's empty and we do this `read`? If I understand correctly, we will get `std::out_of_range("deserialization buffer underflow")` (from `simple_memory_input_stream` implementation). I have to say I liked the previous error better -- "Could not decode the sync point - not a valid hex string".

Reply:
I've restored the `raw.empty()` check.